src/containers/pages/JurisdictionAssignment/EntryView/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { COULD_NOT_LOAD_JURISDICTION, COULD_NOT_LOAD_PLAN } from '../../../../configs/lang';
+import { PlanDefinition } from '../../../../configs/settings';
+import { useLoadingReducer } from '../../../../helpers/useLoadingReducer';
+import { OpenSRPService } from '../../../../services/opensrp';
+import hierarchyReducer, {
+  FetchedTreeAction,
+  fetchTree,
+  reducerName as hierarchyReducerName,
+} from '../../../../store/ducks/opensrp/hierarchies';
+import planDefinitionReducer, {
+  addPlanDefinition,
+  AddPlanDefinitionAction,
+  getPlanDefinitionById,
+  reducerName as planReducerName,
+} from '../../../../store/ducks/opensrp/PlanDefinition';
+import { useHandleBrokenPage } from '../helpers/utils';
+import { ConnectedJurisdictionAssignmentReRouting } from './JurisdictionAssignmentReRouting';
+import { useGetJurisdictionTree, useGetRootJurisdictionId, usePlanEffect } from './utils';
+
+reducerRegistry.register(planReducerName, planDefinitionReducer);
+reducerRegistry.register(hierarchyReducerName, hierarchyReducer);
+
+/** props for this component */
+interface EntryViewProps {
+  plan: PlanDefinition | null;
+  serviceClass: typeof OpenSRPService;
+  treeFetchedCreator: ActionCreator<FetchedTreeAction>;
+  fetchPlanCreator: ActionCreator<AddPlanDefinitionAction>;
+}
+
+const defaultProps = {
+  fetchPlanCreator: addPlanDefinition,
+  plan: null,
+  serviceClass: OpenSRPService,
+  treeFetchedCreator: fetchTree,
+};
+
+/** this component must be served on a route with a planId param */
+export interface EntryViewRouteParams {
+  planId: string;
+}
+
+export type FullEntryViewProps = EntryViewProps & RouteComponentProps<EntryViewRouteParams>;
+
+/** makes service calls for getting the plan and the hierarchy map
+ * and renders component to route to either auto-selection or manual selection
+ */
+const EntryView = (props: FullEntryViewProps) => {
+  const { serviceClass, plan, treeFetchedCreator, fetchPlanCreator } = props;
+
+  const { errorMessage, handleBrokenPage, broken } = useHandleBrokenPage();
+  const initialLoadingState = !plan;
+  const { startLoading, stopLoading, loading } = useLoadingReducer(initialLoadingState);
+
+  usePlanEffect(
+    props.match.params.planId,
+    plan,
+    serviceClass,
+    fetchPlanCreator,
+    handleBrokenPage,
+    stopLoading,
+    startLoading
+  );
+  const rootJurisdictionId = useGetRootJurisdictionId(
+    plan,
+    serviceClass,
+    handleBrokenPage,
+    stopLoading,
+    startLoading
+  );
+  useGetJurisdictionTree(
+    rootJurisdictionId,
+    startLoading,
+    treeFetchedCreator,
+    stopLoading,
+    handleBrokenPage
+  );
+
+  if (loading()) {
+    return <Ripple />;
+  }
+  if (!plan) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_PLAN} />;
+  }
+
+  if (!rootJurisdictionId) {
+    return <ErrorPage errorMessage={COULD_NOT_LOAD_JURISDICTION} />;
+  }
+
+  if (broken) {
+    return <ErrorPage errorMessage={errorMessage} />;
+  }
```

Comment:
These three if blocks have been repeated in at least two other components.  Is it possible to make this more DRY?

Reply:
We can also include the loading if block in that DRYness question

Reply:
thinking about this:
- the if statements are required for a correct program flow. Typescript would also start complaining since it would now not be able to infer that say `plan` cannot be undefined by the time we get to render. 
- Also If these was abstracted to a different component we would 
   - still have to do a conditional check to know when to render that component as opposed to 
the other JSX belonging to the components.

Reply:
cool, lets keep them for now